sdk/search/search-documents/src/searchIndexingBufferedSenderImpl.ts

```diff
@@ -78,7 +81,8 @@ class SearchIndexingBufferedSenderImpl<T> implements SearchIndexingBufferedSende
     this.batchObject = new IndexDocumentsBatch<T>();
     if (this.autoFlush) {
       const interval = setInterval(() => this.flush(), this.flushWindowInMs);
-      interval?.unref();
+      const unblock = interval?.unref;
+      unblock();
```

Comment:
@abc516 Can you please share which linting rule is the above change fixing?

Reply:
@ramya-rao-a it fixes the **no-unused-expressions** rule: https://eslint.org/docs/rules/no-unused-expressions .

Reply:
I feel like we should ignore this one, since the whole point of `?.` is to allow for easy in-line optional access. I don't see a lint option to specifically exclude `?.` from the linter though

Reply:
Hi @xirzec @ramya-rao-a so should I leave this as it is now in the commit, or change it back to the interval?.unref() ?

Reply:
I would like to revert this change.

Reply:
Hi @xirzec @ramya-rao-a the latest commits should address this